trollflow2/plugins/__init__.py

```diff
@@ -189,9 +190,15 @@ def prepared_filename(fmat, renames):
 
     # tmp filenaming
     use_tmp_file = fmat.get('use_tmp_file', False)
+    use_tmp_dir = fmat.get("use_tmp_dir", False)
 
     if use_tmp_file:
-        filename = _get_temp_filename(directory, renames.keys())
+        if use_tmp_dir:
+            tmp_dir = pathlib.Path(use_tmp_dir)
+            of = pathlib.Path(orig_filename)
+            filename = os.fspath(tmp_dir / of.name)
```

Comment:
Do I understand this correctly that the temporary file will not have a temporary name?

Reply:
In which case I would recommend to have this option separated from the the `use_tmp_file` option

Reply:
Agree with @mraspaud. I think the behaviour should be something like this:
```python
if use_tmp_file:
    filename = _get_temp_filename(...)
elif use_tmp_dir:
    filename = os.path.join(use_tmp_dir, os.path.basename(orig_filename))
else:
    filename = orig_filename
```
And as the `use_tmp_dir` is used as the path, it should be renamed as `tmp_dir` or something.

Reply:
It is correct that the temporary file will not have a temporary name.  We need to retain the filename, or find a different solution to the issue described in #125.

I am not convinced that I agree with the separation.  A temporary file in a temporary directory is still a temporary file, it's just implemented differently.  Therefore, I don't think it makes sense to have `use_tmp_dir` without `use_tmp_file`.

Reply:
ok, I can see a point in that, however the configuration needs to be clarified imo. So could we have something like this in the config:
```yaml
    use_tmp_file:
      - filepattern: {platform}_{product}  # should default to a random name like we had before
      - directory: /my/tmp/dir/ # should default to tempfile.gettempdir(), accepts "."
```
?

Reply:
Thinking about this again, maybe it shouldn't be called `tmp_dir` at all, but something like `staging_zone`.  `tmp_dir` suggests a temporary directory is being created, but this is not the case, nor is it necessary.

Reply:
@mraspaud @pnuu I have changed the logic.  The option is now called `staging_zone` and operates independently of `use_tmp_file` (although I'm not sure of a use case for using both, I have covered this case in the new unit test too).